ircd/s_newconf.c

```diff
@@ -685,23 +685,39 @@ time_t
 valid_temp_time(const char *p)
 {
 	time_t result = 0;
-
-	while(*p)
-	{
-		if(IsDigit(*p))
-		{
-			result *= 10;
-			result += ((*p) & 0xF);
-			p++;
-		}
-		else
+	int current = 0;
```

Comment:
Wouldn't long be better, since that's what strtol returns? (Or unsigned long and strtoul?)